### ISSUE TITLE
Use zoofs for regression problems?

### ISSUE THREAD

**Issue by hanamthang** [jaswinder9051998/zoofs#5]
Hi there,

Is there any chance to use zoofs for regression problems?

Many thanks,
Thang

**Comment by jaswinder9051998** [comment]
hi @hanamthang 
 Yes zoofs can be used for regression problems

Feature selection ( what zoofs does ) is basically a subset selection problem, so it doesn't matter if your base task is regression or classification

An example code extract is pasted below, in which we are calculating mean_squared_error using y_valid and X_valid.
You can go ahead and choose any other metric as well, all you need to do is change your self-defined 'objective_function_topass'
```
from sklearn.metrics import mean_squared_error
# define your own objective function, make sure the function receives four parameters,
#  fit your model and return the objective value !
def objective_function_topass(model,X_train, y_train, X_valid, y_valid):      
    model.fit(X_train,y_train)  
    P=mean_squared_error(y_valid,model.predict(X_valid))
    return P

# import an algorithm !  
from zoofs import ParticleSwarmOptimization
# create object of algorithm
algo_object=ParticleSwarmOptimization(objective_function_topass,n_iteration=20,
                                       population_size=20,minimize=True)
import lightgbm as lgb
lgb_model = lgb.LGBMClassifier()                                       
# fit the algorithm
algo_object.fit(lgb_model,X_train, y_train, X_valid, y_valid,verbose=True)
#plot your results
algo_object.plot_history()
```

**Comment by hanamthang** [comment]
Much appreciate to your quick response. 

I tested your codes with my data using the CatBoost and it worked very well. We would like to confirm the performance in next paper with the GWO using your codes. So:

- How can I cite "zoofs" in the paper?
- Do you have any plans for additional algorithms like Harris Haws Optimization - HHO?
- Will you intend to maintain this repository in the future? If possible, I will recommend zoofs for further testing in our group. 
Many thanks,
Thang

**Comment by jaswinder9051998** [comment]
Hi @hanamthang 
You can use the below pasted Bibtex format for citation
```
@misc{zoofs,
  author = {Jaswinder Singh.},
  doi = {10.5281/zenodo.5512921},
  title = {zoofs},
  year = {2021},
  publisher = {GitHub},
  url= {https://github.com/jaswinder9051998/zoofs}},
}
```
we do intend to expand on other algorithms, the development of zoofs relies on contributions as well.

As for your last point, what exactly do you mean by 'further testing in our group' and 'our group', kindly elaborate
